pkg/apis/autoscaling/annotation_validation.go

```diff
@@ -70,6 +71,22 @@ func validateClass(annotations map[string]string) *apis.FieldError {
 	return nil
 }
 
+func validateAlgorithm(annotations map[string]string) *apis.FieldError {
+	if a := annotations[MetricAggregationAlgorithmKey]; a != "" {
+		// Not a KPA? Don't validate, custom autoscalers might have custom values.
+		if c := annotations[ClassAnnotationKey]; c != KPA {
```

Comment:
might as well do this at top-level of method and avoid the nesting

Reply:
I  thought one way or another, felt the same, but sure.